File: clients/dart/lib/src/zawgyi_converter.dart
import 'transliterate/transliterate.dart';
import 'transliterate/transliterate_u2z.dart';
import 'transliterate/transliterate_z2u.dart';
import 'transliterate/transliterate_znorm.dart';

class ZawGyiConverter{

  Transliterate? _transliterateU2Z;
  Transliterate? _transliterateZ2U;
  Transliterate? _transliterateZNorm;

  ///Convert Zawgyi to Unicode
  /// * @param inString Zawgyi string.
  /// * @return Unicode string.
  /// <p> ဇော်ဂျီ မှ ယူနီကုတ် သို့ပြောင်းရန်။
  /// * @param inString ဇော်ဂျီ အချက်အလက်စာ။
  /// * @return ယူနီကုတ် သို့‌ပြောင်းထားသော အချက်အလက်စာ။
  String zawGyiToUnicode(String inString){
    _transliterateZ2U??=TransliterateZ2U('Z2U');
    return _transliterateZ2U!.convert(inString);
  }

  ///Convert Unicode To Zawgyi
  /// * @param inString Unicode string.
  /// * @return Zawgyi string.
  /// <p>ယူနီကုတ် မှ ဇော်ဂျီ သို့ပြောင်းရန်။
  /// * @param inString ယူနီကုတ် အချက်အလက်စာ။
 /// * @return ဇော်ဂျီ သို့‌ပြောင်းထားသော အချက်အလက်စာ။
  String unicodeToZawGyi(String inString){
    _transliterateU2Z??=TransliterateU2Z('U2Z');
    return _transliterateU2Z!.convert(inString);
  }

  ///Convert normalize Zawgyi
  /// * @param inString Zawgyi string.
  /// * @return Normalized Zawgyi string.
  /// <p>ရှုပ်ထွေးသော ဇော်ဂျီ အချက်အလက်စာကြမ်းများ ကို ရှင်းလင်းခြင်း။
  /// * @param  ရှုပ်ထွေးသော ဇော်ဂျီ အချက်အလက်စာကြမ်း။
  /// * @return ရှင်းလင်း ထားသော ဇော်ဂျီ အချက်အလက်စာ။
  String normalizeZawGyi(String inString){
    _transliterateZNorm??=TransliterateZNorm('ZNorm');
    return _transliterateZNorm!.convert(inString);
  }
}

File: clients/c#/MyanmarTools/ZawgyiConverter.cs
using MyanmarTools.Transliterate;

namespace MyanmarTools
{
    public class ZawgyiConverter
    {

        private TransliterateZ2U TransliterateZ2U;
        private TransliterateU2Z TransliterateU2Z;
        private TransliterateZNorm TransliterateZNorm;

        /// <summary>
        ///  <para>Convert Zawgyi to Unicode </para>
        ///  <para>ဇော်ဂျီ မှ ယူနီကုတ် သို့ပြောင်းရန်။ </para>
        /// </summary>
        /// <param name="InString"> Zawgyi string. ဇော်ဂျီ အချက်အလက်စာ။</param>
        /// <returns>Unicode string. ယူနီကုတ် သို့‌ပြောင်းထားသော အချက်အလက်စာ။</returns>
        public string ZawgyiToUnicode(string InString)
        {
            TransliterateZ2U ??= new TransliterateZ2U("Z2U");
            return TransliterateZ2U.Convert(InString);
        }

        /// <summary>
        ///  <para>Convert Unicode To Zawgyi</para>
        ///  <para>ယူနီကုတ် မှ ဇော်ဂျီ သို့ပြောင်းရန်။</para>
        /// </summary>
        /// <param name="InString">Unicode string. ယူနီကုတ် အချက်အလက်စာ။</param>
        /// <returns>Zawgyi string.ဇော်ဂျီ သို့‌ပြောင်းထားသော အချက်အလက်စာ။ </returns>
        public string UnicodeToZawgyi(string InString)
        {
            TransliterateU2Z ??= new TransliterateU2Z("U2Z");
            return TransliterateU2Z!.Convert(InString);
        }

        /// <summary>
        /// <para>Convert normalize Zawgyi</para>
        /// <para>ရှုပ်ထွေးသော ဇော်ဂျီ အချက်အလက်စာကြမ်းများ ကို ရှင်းလင်းခြင်း။</para>
        /// </summary>
        /// <param name="InString">Zawgyi string.  ရှုပ်ထွေးသော ဇော်ဂျီ အချက်အလက်စာကြမ်း။</param>
        /// <returns>Normalized Zawgyi string.  ရှင်းလင်း ထားသော ဇော်ဂျီ အချက်အလက်စာ။</returns>
        public string NormalizeZawgyi(string InString)
        {
            TransliterateZNorm ??= new TransliterateZNorm("ZNorm");
            return TransliterateZNorm!.Convert(InString);
        }
    }
}